File: string/string.js
// implement string using ES6 class syntax
class MyString {
  constructor() {
    this.string = '';
  }

  // is string empty
  isEmpty() {
    if (this.string === '') return true;
    else return false;
  }

  // add a character to a string
  add(ch) {
    return this.string += ch;
  } 

  // print string
  print() {
    console.log(this.string);
  }

  // convert string to lower case
  lower() {
    if (this.isEmpty()) return "";
    else return this.string.toLowerCase();
  }

  // convert string to upper case
  upper() {
    if (this.isEmpty()) return "";
    else return this.string.toUpperCase();
  }

  // whether a character in a string
  hasCh(ch) {
    if (this.isEmpty()) return "";
    else {
      for (let i = 0; i < this.string.length; i += 1) {
        if (this.string[i] === ch) return i;
      }
      return -1;
    }
  }
}

File: string/string.py
from typing import Optional

class MyString:
	'''
	implement string using python
	'''
	def __init__(self) -> None:
		self.string: str = ""

	def is_empty(self) -> bool:
		'''
		is string empty
		'''
		if self.string == "":
			return True
		else:
			return False

	def add(self, ch: str) -> str:
		'''
		add a character to a string
		'''
		self.string += ch
		return self.string

	def print(self) -> None:
		'''
		print string
		'''
		print(self.string)

	def lower(self) -> str:
		'''
		convert string to lower case
		'''
		if self.is_empty():
			return ""
		else:
			return self.string.lower()

	def upper(self) -> str:
		'''
		convert string to upper case
		'''
		if self.is_empty():
			return ""
		else:
			return self.string.upper()

	def has_ch(self, ch: str) -> Optional[int]:
		'''
		whether a character in a string
		'''
		if self.is_empty():
			return False
		else:
			for i in range(len(self.string)):
				if self.string[i] == ch:
					return i
			return -1